schema/home/home_page_artist_module.js

```diff
@@ -7,40 +7,48 @@ import gravity from '../../lib/loaders/gravity';
 import { NodeInterface } from '../object_identification';
 import { toGlobalId } from 'graphql-relay';
 import {
+  GraphQLEnumType,
   GraphQLID,
   GraphQLNonNull,
   GraphQLList,
   GraphQLObjectType,
   GraphQLString,
 } from 'graphql';
 
-export const Results = {
-  suggested: {
+// This object is used for both the `key` argument enum and to do fetching.
+export const HomePageArtistModuleTypes = {
+  SUGGESTED: {
+    description: 'Artists recommended for the specific user.',
     fetch: (accessToken, userID) => {
       return gravity.with(accessToken)(`user/${userID}/suggested/similar/artists`);
     },
     display: (accessToken, userID) => {
+      const fetch = HomePageArtistModuleTypes.SUGGESTED.fetch;
       return Promise.resolve(!!(accessToken && userID)).then(display => {
-        return display && Results.suggested.fetch(accessToken, userID).then(results => {
+        return display && fetch(accessToken, userID).then(results => {
           return results.length > 0;
```

Comment:
Ok, vastly simplified by using a conditional and returning different promises depending on if the user has access or not.